src/annotator/highlighter.ts

```diff
@@ -442,11 +442,22 @@ export function getHighlightsFromPoint(
   x: number,
   y: number,
 ): HighlightElement[] {
-  return document
-    .elementsFromPoint(x, y)
-    .filter(
-      el => el.localName === 'hypothesis-highlight',
-    ) as HighlightElement[];
+  // We test each `<hypothesis-highlight>` element rather than using
+  // `document.elementsFromPoint` here because shape highlights have
+  // `pointer-events: none` set on them, so that users can still interact with
+  // the content underneath. This has a side effect of making the elements
+  // invisible to hit-testing methods like `elementsFromPoint`.
+  //
+  // The downside is that we're iterating through all highlights every time
+  // this is called, which might get expensive if there are a lot.
```

Comment:
I have identified a performance hazard here, but I think the cost should be low enough that we can use this simple code until we have proven there is a problem and need to do something more elaborate.

Reply:
Perhaps a quick performance improvement would involve not unpacking `[...document.querySelectorAll('hypothesis-highlight')]` and `[...el.getClientRects()]`, which iterates them entirely and then again for filtering, and instead iterate them with regular loops and filter or match manually.

It's more verbose, but it might help here.